core/server/api_container/server/startosis_engine/kurtosis_instruction/shared_helpers/service_helpers.go

```diff
@@ -25,47 +24,63 @@ func ExecuteServiceAssertionWithRecipe(
 	interval time.Duration,
 	timeout time.Duration,
 ) (map[string]starlark.Comparable, int, error) {
-	var requestErr error
+	var recipeErr error
 	var assertErr error
 	tries := 0
-	timedOut := false
 	lastResult := map[string]starlark.Comparable{}
-	startTime := time.Now()
-
-	backoffObj := backoff.NewConstantBackOff(interval)
+	deadline := time.Now().Add(timeout)
+	contextWithDeadline, cancelContext := context.WithDeadline(ctx, deadline)
+	defer cancelContext()
+	timeoutChan := time.After(timeout)
+	tickChan := time.Tick(interval)
```

Comment:
in addition to the linter, can we make sure we defer a Stop those timers/tickers?